### ISSUE TITLE
MapField scaffolder doesn't work on Page classes

### ISSUE THREAD

**Issue by phptek** [smindel/silverstripe-gis#2]
If I follow the same, minmal instructions from the README, but use `Page` instead of `DataObject`, I don't see a scaffolded leaflet map in the admin UI, when creating a new `Page`. If I create a new `DataObject`, I see a map, but it has no basemap - the latter problem could be donw to config. I'll play around a little more.

PHP 7.1 / Ubuntu 16.04 / Framework & CMS 4.3 / silverstripe/admin 1.3

EDIT:

```
class PlanPage extends Page
{
    private static $db = [
        'Location' => 'Geometry',
    ];

    public function getCMSFields()
    {
        ...
        $fields->dataFieldByName('Location'); // null
        ...
    }
    ...
}
```

EDIT: basemap appears after saving - see #4

**Comment by smindel** [comment]
Since SiteTree doesn't use the scaffolder, you have to manually add the field to your custom page type in MyPage::getCMSFields() like this:

`
public function getCMSFields()
{
    parent::getCMSFields()->addFieldToTab('Root.Main', MapField::create('MyLocation'));
}
`

**Comment by smindel** [comment]
The MapField breaking when the value is empty has been fixed in 1.1.1
